spec/support/feature_spec_helpers.rb

```diff
@@ -25,6 +25,10 @@ def sign_in(user)
     end
 
     goto_page HomePage do |page|
+      unless page.has_form?
+        page.user_menu.icon.click
+        page.user_menu.logout.click
+      end
```

Comment:
Did you run into an issue to sign_in while already sign_in? What about an explicit sign_out or maybe testing the currently loaded URL instead?

Reply:
Yes, if a user is logged in, there is no login form. We need to log out first.

I think this is fine. This is a high level action to sign in a user. The `sign_in` helper takes care of that. When I'm logged in and I tell you to sign in, it naturally requires to sign out first.
Do you see any downside to this implicit approach?